src/Workspaces/Core/Portable/SymbolKey/SymbolKey.AnonymousTypeSymbolKey.cs

```diff
@@ -29,18 +29,18 @@ public static void Create(INamedTypeSymbol symbol, SymbolKeyWriter visitor)
 
             public static SymbolKeyResolution Resolve(SymbolKeyReader reader)
             {
-                var propertyTypeSymbols = reader.ReadSymbolKeyArray();
-                var propertyTypes = propertyTypeSymbols.Select(r => GetFirstSymbol<ITypeSymbol>(r)).ToImmutableArray();
-                var propertyNames = reader.ReadStringArray();
-                var propertyIsReadOnly = reader.ReadBooleanArray();
-                var propertyLocations = reader.ReadLocationArray();
+                using var propertyTypes = reader.ReadSymbolArray<ITypeSymbol>();
+                using var propertyNames = reader.ReadStringArray();
+                using var propertyIsReadOnly = reader.ReadBooleanArray();
+                using var propertyLocations = reader.ReadLocationArray();
```

Comment:
arrays during resolutoin are pooled so that we can we don't waste memory on them.